core/bit_stream/bitstream.odin

```diff
@@ -0,0 +1,116 @@
+
+package bs;
+
+import "core:mem"
+
+BITSTREAM_INITIAL_SIZE: int: 128;
+BITSTREAM_GROWTH_RATE:  f64: 2.0;
+
+Bit_Stream :: struct {
+    data, head: ^byte,
+    size, cap: int,
```

Comment:
I would recommend using a slice for the data rather than a pointer. 
```
data: []byte,
head: int,
len: int,
```

Reply:
But using pointers may not be an issue in practice if it is only meant to be used internally.

Reply:
I chose for using pointers because it makes a lot of memory operations trivial.
The original use of this package was to construct network packets for a game so I would need the pointer and the size for that and I felt that having a slice would add unnecessary friction.

I'll update the package to remove the unnecessary `auto_cast`s and change the structure to something like you described sometime this week (probably).